eppo_metrics_sync/dbt_model_parser.py

```diff
@@ -0,0 +1,128 @@
+                        "description": column["description"]
+                    })
+                
+                if 'eppo_property' in tags:
+                    self.eppo_properties.append({
+                        "name": column["name"],
+                        "column": column["name"],
+                        "description": column["description"]
+                    })
+    
+    def validate(self):
+
+        self.validation_errors = []
+
+        # make sure there is at least one entity and exactly one timestamp
+        if len(self.eppo_timestamp) == 0:
+            self.validation_errors.append(
+                'Exactly 1 column must be have tag "eppo_timestamp"'
+            )
+        
+        if len(self.eppo_entities) == 0:
+            self.validation_errors.append(
+                'At least 1 column must have tag "eppo_entity:<entity_name>"'
+            )
+        
+        # check that no columns are tagged to multiple Eppo fields (skip if there
+        # is already a validation error)
+
+        if len(self.validation_errors) == 0:
+            # parse names from column list
+            entity_names = [e['column'] for e in self.eppo_entities]
+            timestamp_names = [self.eppo_timestamp]
+            fact_names = [e['column'] for e in self.eppo_facts]
+            property_names = [e['column'] for e in self.eppo_properties]
+
+            all_names = Counter(chain.from_iterable(
+                [entity_names, timestamp_names, fact_names, property_names]
+            ))
+
+            overlapping_columns = [item for item, count in all_names.items() if count > 1]
+
+            if len(overlapping_columns):
+                self.validation_errors.append(
+                    f'The following columns had tags to multiple Eppo fields: {", ".join(overlapping_columns)}'
+                )
+            
+        
+        if len(self.validation_errors):
+            raise ValueError(
+                f'One or more errors parsing model: {self.model["name"]}: {", ".join(self.validation_errors)}'
+            )
+
+    
+    def format(self):
+
+        entity_clause = '\n  , '.join([e['column'] for e in self.eppo_entities])
+        fact_clause = '\n  , '.join([f['column'] for f in self.eppo_facts])
+        property_clause = '\n  , '.join([p['column'] for p in self.eppo_properties])
+
+        formatted_sql = \
+         f"""
+        select
+            {self.eppo_timestamp}
+            , {entity_clause}
+            , {fact_clause}
+            , {property_clause}
+        from 
+            {self.dbt_model_prefix}.{self.model['name']}
+        """
+
+        self.eppo_fact_source = {
+            "name": self.model["name"],
+            "sql": formatted_sql,
+            "timestamp_column": self.eppo_timestamp,
+            "entities": self.eppo_entities,
+            "facts": self.eppo_facts,
+            "properties": self.eppo_properties
+        }
+    
+    def build(self):
```

Comment:
Love this! It's a great way of conveying to the reader that there are three "high-level" steps in the build job, but the details of those steps are abstracted away to other dedicated methods.